nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AbstractQueryDatabaseTable.java

```diff
@@ -343,6 +362,24 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
             if (logger.isDebugEnabled()) {
                 logger.debug("Executing query {}", new Object[] { selectQuery });
             }
+
+            final boolean originalAutoCommit = con.getAutoCommit();
+            final Boolean propertyAutoCommitValue = context.getProperty(AUTO_COMMIT).evaluateAttributeExpressions().asBoolean();
+            // If user sets AUTO_COMMIT property to non-null (i.e. true or false), then the property value overrides the dbAdapter's value
+            final Boolean setAutoCommitValue =
+                    dbAdapter == null || propertyAutoCommitValue != null
+                            ? propertyAutoCommitValue
+                            : dbAdapter.getAutoCommitForReads(fetchSize).orElse(null);
+            if (setAutoCommitValue != null && originalAutoCommit != setAutoCommitValue) {
```

Comment:
We should add logic to customValidate to check  the property setting against the adapter setting and mark the processor invalid if they don't match (if Fetch Size is non-zero). It's very little risk to current users because their current flows wouldn't work correctly anyway.